### PR TITLE
iOS-2335 Space Manager | Show my permissions

### PR DESCRIPTION
**Pull request by mgolovko**

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/Buttons/AsyncStandardButton.swift [hunk 1]
```diff
@@ -16,7 +17,9 @@ struct AsyncStandardButton: View {
             Task {
                 do {
                     try await action()
-                } catch {}
+                } catch {
+                    toast = ToastBarData(text: error.localizedDescription, showSnackBar: true, messageType: .failure)
+                }
```
Comment (ignatovv) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/Buttons/AsyncStandardButton.swift:22]: 💪

File: Anytype/Sources/PresentationLayer/Modules/SpacesManager/SpacesManagerViewModel.swift [hunk 2]
```diff
@@ -5,21 +5,43 @@ import Combine
...
+            let participant = participants.first { $0.spaceId == spaceView.targetSpaceId }
+            return SpacesManagerRowViewModel(
+                id: spaceView.id,
+                name: spaceView.name,
+                iconImage: spaceView.iconImage,
+                accountStatus: spaceView.accountStatus?.name ?? "",
+                localStatus: spaceView.localStatus?.name ?? "",
+                permission: participant?.permission.title ?? ""
+            )
+        }
```
Comment (ignatovv) [inline Anytype/Sources/PresentationLayer/Modules/SpacesManager/SpacesManagerViewModel.swift:45]: Mb create dedicated extension init(spaceView:, participant:) ?

File: Anytype/Sources/ServiceLayer/ServiceLocator.swift [hunk 3]
```diff
@@ -209,8 +209,15 @@ final class ServiceLocator {
...
-    func participantSubscriptionService() -> ParticipantsSubscriptionServiceProtocol {
-        return ParticipantsSubscriptionService(
+    func participantSubscriptionServiceBySpace() -> ParticipantsSubscriptionServiceBySpaceProtocol {
+        return ParticipantsSubscriptionServiceBySpace(
+            subscriptionStorageProvider: subscriptionStorageProvider(),
+            activeWorkspaceStorage: activeWorkspaceStorage()
+        )
+    }
+    
+    func participantsSubscriptionServiceByAccount() -> ParticipantsSubscriptionServiceByAccountProtocol {
```
Comment (ignatovv) [inline Anytype/Sources/ServiceLayer/ServiceLocator.swift:219]: participantsSubscriptionServiceByAccount -> participantsSubscriptionByAccountService
participantSubscriptionServiceBySpace -> participantSubscriptionBySpaceService
more human readable imho

**Review by ignatovv — APPROVED** [review]

File: Anytype/Sources/ServiceLayer/ServiceLocator.swift (hunk 3 above)
Comment (mgolovko) [reply Anytype/Sources/ServiceLayer/ServiceLocator.swift]: Fixed

File: Anytype/Sources/PresentationLayer/Modules/SpacesManager/SpacesManagerViewModel.swift (hunk 2 above)
Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/Modules/SpacesManager/SpacesManagerViewModel.swift]: Fixed

**Pull request merged**